File: library/src/test/java/io/ashdavies/rx/rxtasks/CompletableEmitterFailureListenerTest.java
package io.ashdavies.rx.rxtasks;

import io.reactivex.CompletableEmitter;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static org.mockito.BDDMockito.then;
import static org.mockito.Mockito.never;

@RunWith(MockitoJUnitRunner.class)
public class CompletableEmitterFailureListenerTest {

  private CompletableEmitterFailureListener listener;

  @Mock CompletableEmitter emitter;
  @Mock Exception exception;

  @Before
  public void setUp() throws Exception {
    listener = new CompletableEmitterFailureListener(emitter);
  }

  @Test
  public void shouldCallOnErrorWithException() throws Exception {
    listener.onFailure(exception);

    then(emitter).should(never()).onComplete();
    then(emitter).should().onError(exception);
  }
}

File: library/src/test/kotlin/io/ashdavies/rx/rxtasks/CompletableEmitterFailureListenerTest.kt
package io.ashdavies.rx.rxtasks

import com.nhaarman.mockito_kotlin.mock
import io.reactivex.CompletableEmitter
import org.junit.Test
import org.mockito.BDDMockito.then
import org.mockito.Mockito.never
import java.lang.Exception

internal class CompletableEmitterFailureListenerTest {

  private val emitter = mock<CompletableEmitter>()
  private val exception = mock<Exception>()

  private val listener = CompletableEmitterFailureListener(emitter)

  @Test
  fun `should call on error with exception`() {
    listener.onFailure(exception)

    then(emitter).should(never()).onComplete()
    then(emitter).should().onError(exception)
  }
}
